### ISSUE TITLE
Some way to conveniently alter hit chance of a melee weapon in json with enchantments or relic_data

### ISSUE THREAD

**Issue by ampersand55** [CleverRaven/Cataclysm-DDA#77983]
### Is your feature request related to a problem? Please describe.

You can alter most other things such as dispersion of a ranged weapon, attack speed, general damage, specific damage, melee stamina consumption etc with `enchantments` or `relic_data`, example:

```json
    "enchantments": [
      {
        "condition": "WIELD",
        "values": [
          { "value": "MELEE_DAMAGE", "add": 10 },
          { "melee_damage_bonus": [ { "type": "bash", "add": 10 } ] }
        ]
      }
    ]
```

But for hit chance its more complicated. One way is to have the item transform into another item with `"relative": { "to_hit": 1 }`, but that uses the legacy `to_hit` value instead of the preferred new way with grip, length, surface and balance. That doesn't work if you want it applied to an arbitrary weapon. 

A way used by magiclysm's item "pair of bracers of archery" is to create an `effect_type` with a `hit mod` that gets applied with an a relic_data passive effect, but that seem like a rather roundabout way.

### Solution you would like.

Something like an enchantment value that looks like this:

```json
          { "value": "HIT_CHANCE", "add": 1},
```

https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/MAGIC.md#id-values

**Comment by Procyonae** [comment]
While this is easy enough to add your use case seems dumb, like if we take one of the +1 weapons from Magiclysm
```
  {
    "type": "GENERIC",
    "id": "warhammer_plus_one",
    "copy-from": "warhammer",
    "name": { "str": "warhammer +1", "str_pl": "warhammers +1" },
    "proportional": { "price": 3.0, "price_postapoc": 3.0, "weight": 0.9, "melee_damage": { "all": 1.1 } },
    "relative": { "to_hit": 1 }
  },
```
and replace it with an enchantment via relic data it looks like
```
  {
    "type": "GENERIC",
    "id": "warhammer_plus_one",
    "copy-from": "warhammer",
    "name": { "str": "warhammer +1", "str_pl": "warhammers +1" },
    "proportional": { "price": 3.0, "price_postapoc": 3.0, "weight": 0.9, "melee_damage": { "all": 1.1 } },
    "relic_data": { "passive_effects": [ { "id": "+1_to_hit" } ] }
  },
  {
    "type": "enchantment",
    "id": "+1_to_hit",
    "has": "WIELD",
    "condition": "ALWAYS",
    "values": [ { "value": "MELEE_TO_HIT", "add": 1 } ]
  },
```
Which is the same functionality just in a more ugly way (Factoring in that the enchantment would be reused for all the +1 weapons) and also isn't really any better than the effect method mentioned.

I don't really understand why you think using relative with to_hit is using the "old method", it'd be pretty weird if we made it so you had to replace the individual named stats when most of the uses in repo are for some form of magical nonsense not improving the grip or whatever and idk how else you'd suggest it be changed.

TLDR I'm adding this but please don't use it in the way you outlined imo

**Comment by ampersand55** [comment]
> I don't understand your use case, like if we take one of the +1 weapons from Magiclysm:

I don't think the Magiclysm war hammer  `"relative": { "to_hit": 1 }` even work, as it's copied from the regular war hammer which gets its hit value calculated from:

```
    "to_hit": { "grip": "weapon", "length": "long", "surface": "line", "balance": "good" },
```

I think the legacy added hit chance added by `"relative": { "to_hit": 1 }` isn't used at all, or gets overwritten by new the hit chance system. 

I tried changing the Magiclysm war hammer to `"relative": { "to_hit": 10 }`, but its hit chance didn't increase, at least not in the info screen.

Anyway, thanks for doing this!
